libqsopt/CMakeLists.txt

```diff
@@ -20,17 +23,29 @@ if(CMAKE_DPKG)
   endif()
 endif()
 
-
 if(SUPPORTED MATCHES "true")
+  file(WRITE ${CATKIN_DEVEL_PREFIX}/${CATKIN_PACKAGE_INCLUDE_DESTINATION}/qsopt.h "")  # create dummy file for correct configuration of libconcorde_tsp_solver
+
   catkin_package(
+    INCLUDE_DIRS
+      common/lib/${MYARCH}
     CFG_EXTRAS libqsopt-extras.cmake # catkin_package does not support artifacts in devel space
   )
 
-  install(FILES common/lib/${MYARCH}/qsopt.a
-    DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+  # Concorde expects qsopt.a to be in the same directory as qsopt.h --> static library, when using in another package one has to provide an absolut path to it
+  add_custom_target(
+    build_qsopt ALL
+      # copy headers to devel space (catkin does not like headers in source space)
+      COMMAND ${CMAKE_COMMAND} -E copy_if_different ${PROJECT_SOURCE_DIR}/common/lib/${MYARCH}/qsopt.h ${CATKIN_DEVEL_PREFIX}/${CATKIN_PACKAGE_INCLUDE_DESTINATION}
+      # copy static lib to include destination as well because concorde needs both files in the same folder
+      COMMAND ${CMAKE_COMMAND} -E copy_if_different ${PROJECT_SOURCE_DIR}/common/lib/${MYARCH}/qsopt.a ${CATKIN_DEVEL_PREFIX}/${CATKIN_PACKAGE_INCLUDE_DESTINATION}
+  )
+
+  install(DIRECTORY ${CATKIN_DEVEL_PREFIX}/${CATKIN_PACKAGE_INCLUDE_DESTINATION}/
+    DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
   )
 
-  install(FILES common/lib/${MYARCH}/qsopt.h
+  install(FILES ${CATKIN_DEVEL_PREFIX}/${CATKIN_PACKAGE_INCLUDE_DESTINATION}/qsopt.a
```

Comment:
Why is a library being installed into the include directory?

Reply:
@ipa-fez 
it's not install **to** but installed **from** an include directory...
thus [this line](https://github.com/ipa320/cob_extern/pull/91/files#diff-6fc75a205d653a1a7d033c8edf2f26e5R41) is already not correct

Reply:
I have to copy the .a file into the same folder as the .h file because the software binary using this library requires this. You may choose whether both appear in include or lib, but somewhere they both need to go.

Reply:
sounds strange...maybe you need to modify/extend https://github.com/ipa-rmb/cob_extern/blob/indigo_dev/libqsopt/cmake/libqsopt-extras.cmake.in?

Reply:
Thing is, I wonder whether this change still passes the OSRF build farm...because it looks hacky...and the farm is much stricter...than out TravisCI